src/test/java/bio/overture/ego/grpc/service/UserServiceGrpcAuthTest.java

```diff
@@ -0,0 +1,269 @@
+            .addService(ServerInterceptors.intercept(userServiceGrpc, authInterceptor))
+            .build()
+            .start());
+
+    stub = UserServiceGrpc.newBlockingStub(channel);
+  }
+
+  @Before
+  public void testDataSetup() {
+    if (!hasRunSetup) {
+      hasRunSetup = true;
+      // Setup test data and meta data for auth
+      testUser =
+          userService.create(
+              CreateUserRequest.builder()
+                  .status(StatusType.APPROVED)
+                  .email("approvedUserGrpc@example.com")
+                  .type(UserType.USER)
+                  .build());
+      userAuthMeta.put(JWT_KEY, tokenService.generateUserToken(testUser));
+
+      testAdmin =
+          userService.create(
+              CreateUserRequest.builder()
+                  .status(StatusType.APPROVED)
+                  .email("approvedAdminGrpc@example.com")
+                  .type(UserType.ADMIN)
+                  .build());
+      userAdminAuthMeta.put(JWT_KEY, tokenService.generateUserToken(testAdmin));
+
+      testApp =
+          appService.create(
+              CreateApplicationRequest.builder()
+                  .status(StatusType.APPROVED)
+                  .type(ApplicationType.CLIENT)
+                  .clientId("grpctest")
+                  .clientSecret("grpctestsecret")
+                  .redirectUri("http://test.example.com")
+                  .name("grpctest")
+                  .build());
+      appAuthMeta.put(JWT_KEY, tokenService.generateAppToken(testApp));
+    }
+  }
+
+  @Test
+  public void getUser_noAuth_rejected() {
+
+    val noAuthStub = MetadataUtils.attachHeaders(stub, emptyAuthMeta);
+
+    // Test that the interceptor rejects this request
+    try {
+
+      noAuthStub.getUser(GetUserRequest.newBuilder().setId(UUID.randomUUID().toString()).build());
+      fail("Request should be rejected due to missing JWT");
+    } catch (StatusRuntimeException e) {
+      // Yay happy times it failed as expected
+    }
+  }
+
+  @Test
+  public void getUser_userAuth_success() {
+
+    val authStub = MetadataUtils.attachHeaders(stub, userAuthMeta);
+
+    // Test that the interceptor rejects this request
+    try {
+      val reply =
+          authStub.getUser(GetUserRequest.newBuilder().setId(testUser.getId().toString()).build());
+      assertThat(reply.getId().getValue()).isEqualTo(testUser.getId().toString());
+
+    } catch (StatusRuntimeException e) {
+      fail("User disallowed to access their ego data via grpc.");
+    }
+  }
+
+  @Test
+  public void getUser_userAuth_rejectedForWrongUser() {
+
+    val authStub = MetadataUtils.attachHeaders(stub, userAuthMeta);
+
+    // Test that the interceptor rejects this request
+    try {
+      // Generate a random ID, ensure that it doesn't match our user (almost impossible but check
+      // anyways so we know fo sho)
+      UUID randomId;
+      do {
+        randomId = UUID.randomUUID();
+      } while (randomId.equals(testUser.getId()));
```

Comment:
even through you dont care if it exists or not becuase you only care that it doesnt match the `testUser` id, you could optionally use `val randomId = EntityGenerator.generateNonExistentId(userService);`, which will do the same thing.

Reply:
Didn't know that was available, thanks.